web/src/spoilers.ts

```diff
@@ -64,10 +64,17 @@ export function initialize(): void {
             return;
         }
 
-        // Allow selecting text inside a spoiler header.
-        const selection = document.getSelection();
-        if (selection?.type === "Range") {
-            return;
+        // Allow selecting text inside a spoiler header,
+        const selection = window.getSelection();
+        if (selection?.type === "Range" && selection?.rangeCount > 0) {
+            const $header = $(this);
+            const range = selection.getRangeAt(0);
+            if (
+                $header[0]?.contains(range.commonAncestorContainer) &&
+                !$spoiler_content[0]?.contains(range.commonAncestorContainer)
+            ) {
+                return;
```

Comment:
This algorithm needs a comment explaining the motivation (double clicks, I guess?) and how this logic prevents that failure mode without breaking stuff like selecting the entire message body.